### ISSUE TITLE
review issues

### ISSUE THREAD

**Issue by behinger** [alixlam/fmristroke#21]
I'm giving it a new try, thanks for the fixes and sorry for my bad time-management.
- I'm testing with python 3.11
- on the test-dataset; which is not ideal because it tests with the same dataset, but I dont have another dataset available.

Let's first play naive user, without reading the docs / how-to online
- I really like that you have a proper help for the CLI tool, very cool!!
- I could not find out what this error means:
`fmristroke: error: the following arguments are required: analysis_level`
Here: https://fmristroke.readthedocs.io/en/latest/usage.html I found kind of an explanation - I didnt understand that the {participant} means I should select from the 1-tuple (participants) the only implemented option right now - and that this is referred by analysis_level. This should be easily adressable by just adding two words to the help or so
- ok, now it can't find my bids-folder: `❯ fmristroke FCStroke_2009_deriv out_test participant
` results in
`ValueError: BIDS root does not exist: /store/users/ehinger/peerreview/fmristroke/fmristroke/data/tests/None` - not sure where the `None` comes from? Also "cding" into the folder and running with `./` did not help. Ah I see, the right command is:
`fmristroke FCStroke_2009 out_test participant --participant-label sub-027 --fmriprep_dir=FCStroke_2009_deriv`

Which runs me into my next problem:
`sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked`
Okay, specifying a work directory `-w /tmp/fmristroke` fixed it. 

It is now running and doing things ;)

---

I guess following the documentation proper would have alleviated these problems - given that the docs clearly state to specify
`fmristroke data/bids_root/ out/ participant -w work/ --fmriprep_dir derivatives/` - but maybe my tour still helps 

---

Finally now I understood what you wrote in #14 that this is just dummy data (indeed, now I noticed the files are all empty anyway). I couldnt find any other prepared dataset as you mentioned in https://github.com/alixlam/fmristroke/issues/14#issuecomment-2084615050 

I simply assumed the data are somewhere when you closed the issue, but I cant seem to find any pointers to it.

**Comment by alixlam** [comment]
Hi @behinger ! thanks for your time and your first feedback  !
You can find some test data here:  https://osf.io/5z4ry/ since the files are a bit heavy I did not include them in the repo :)

**Comment by behinger** [comment]
Thanks! On that dataset I run into a new problem:
```bash
          File "/home/ehinger/micromamba/envs/fmristroke/bin/rapidtide", line 23, in <module>
            rapidtide_workflow.rapidtide_main(rapidtide_parser.process_args(inputargs=None))
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/rapidtide/workflows/rapidtide.py", line 425, in rapidtide_main
            raise ValueError(
        ValueError: magnitude of lagmin exceeds 5.0 - invalid
Traceback:
        Traceback (most recent call last):
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/core.py", line 453, in aggregate_outputs
            setattr(outputs, key, val)
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/traits_extension.py", line 330, in validate
            value = super(File, self).validate(objekt, name, value, return_pathlike=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/traits_extension.py", line 135, in validate
            self.error(objekt, name, str(value))
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/traits/base_trait_handler.py", line 74, in error
            raise TraitError(
        traits.trait_errors.TraitError: The 'output_corrmap' trait of a RapidTideOutputSpec instance must be a pathlike object or string representing an existing file, but a value of '/tmp/fmristroke/fmristroke_0_1_wf/single_subject_M2121_wf/lesion_conn_ses_258_task_rest_run_17_space_T1w_desc_preproc_wf/hemodynamic_wf/rapidtide/concat_desc-maxcorr_map.nii.gz' <class 'str'> was specified.

        During handling of the above exception, another exception occurred:

        Traceback (most recent call last):
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/core.py", line 400, in run
            outputs = self.aggregate_outputs(runtime)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/core.py", line 460, in aggregate_outputs
            raise FileNotFoundError(msg)
        FileNotFoundError: No such file or directory '/tmp/fmristroke/fmristroke_0_1_wf/single_subject_M2121_wf/lesion_conn_ses_258_task_rest_run_17_space_T1w_desc_preproc_wf/hemodynamic_wf/rapidtide/concat_desc-maxcorr_map.nii.gz' for output 'output_corrmap' of a RapidTide interface
```
```bash
fmristroke . derivatives/out_test participant --fmriprep_dir=derivatives -w /tmp/fmristroke  
```

**Comment by alixlam** [comment]
This seems similar to an error experienced before by SRSteinkamp. I should have fixed it, did you pull the code recently ?

**Comment by behinger** [comment]
the version was 0.1.0 - I upgraded it maybe ~1 week ago?

I now used
`❯ python -m pip install git+https://github.com/alixlam/fmristroke.git --upgrade`
Resolving to
`  Resolved https://github.com/alixlam/fmristroke.git to commit 47e02c135efe03e9f6cefd5cb12faca79e5386be`

which is the newest one.

rerunning: same result

**Comment by SRSteinkamp** [comment]
Might this be the issue, that the example dataset does not have enough data to run rapitide on? So adding `--session-level` might help here?

**Comment by alixlam** [comment]
Yes it looks like this issue, the run 17 is too short for rapidtide, but I should have added a check of bold size and skips the ones that are too short (and adding a warning) if the --session-level is not on. Will try to run on my side with a clean environment.

**Comment by alixlam** [comment]
![Screenshot 2024-11-08 at 15 47 47](https://github.com/user-attachments/assets/98de92ce-ae23-4293-935a-2d98d20b7ac2)
@behinger do you have a similar output on your terminal, especially the last part ?

**Comment by behinger** [comment]
Nope - but I get a ton of output; how do you run yours with less output?

```bash
241108-14:12:53,780 nipype.workflow IMPORTANT:                                                                                                                                                                                                           
         Running fMRIStroke version 0.1.0                                                                                                                                                                                                                
241108-14:12:54,42 nipype.workflow IMPORTANT:                                                                                                                                                                                                            
         Building fMRIStroke's workflow:                                                                                                                                                                                                                 
           * BIDS dataset path: /scratch/data/fmristroke.                                                                                                                                                                                                
           * fMRIPrep path: /scratch/data/fmristroke/derivatives.                                                                                                                                                                                        
           * Participant list: ['M2112', 'M2121'].                                                                                                                                                                                                       
           * Run identifier: 20241108-141247_5236511b-d6e1-4c16-ba4a-d0e58d1540b0.                                                                                                                                                                       
           * Output spaces: MNI152NLin2009cAsym:res-native.                                                                                                                                                                                              
           * Denoising pipelines: ['SimpleGS', 'SimpleLesionGS', 'ICLesionCompCorGS', 'Minimal'].                                                                                                                                                        
           * Atlases: ['Scheafer'].                                                                                                                                                                                                                      
241108-14:13:00,151 nipype.workflow INFO:                                                                                                                                                                                                                
         fMRIStroke workflow graph with 488 nodes built successfully.                                                                                                                                                                                    
241108-14:13:02,906 nipype.workflow IMPORTANT:                                                                                                                                                                                                           
         fMRIStroke started!                                                                                                                                                                                                                             
241108-14:13:05,925 nipype.workflow INFO:                                                                                                                                                                                                                
         [Node] Setting-up "fmristroke_0_1_wf.single_subject_M2121_wf.lesion_conn_ses_258_task_rest_run_17_space_T1w_desc_preproc_wf.connectivity_wf.atlas_info" in "/tmp/fmristroke/fmristroke_0_1_wf/single_subject_M2121_wf/lesion_conn_ses_258_task_r
un_17_space_T1w_desc_preproc_wf/connectivity_wf/_atlas_Scheaffer/atlas_info".                                                                                                                                                                            
241108-14:13:05,925 nipype.workflow INFO:                                                                                                                                                                                                                
         [Node] Outdated cache found for "fmristroke_0_1_wf.single_subject_M2121_wf.lesion_conn_ses_258_task_rest_run_17_space_T1w_desc_preproc_wf.connectivity_wf.atlas_info".                                                                          
241108-14:13:05,928 nipype.workflow INFO:                                                                                                                                                                                                                

```

**Comment by alixlam** [comment]
It is normal to have a ton of outputs, and I have not implemented a version with less. What is not normal is that you don't have the Warning at the beginning ...

**Comment by alixlam** [comment]
Since I have not changed the version when doing those commit, it is possible that your version is not upgraded. It might be worth uninstalling and reinstalling.

**Comment by behinger** [comment]
indeed, that worked now. Sorry - I simply hate pip for doing that - but I ran into new errors :| 

```python
241108-16:35:01,644 nipype.workflow INFO:                                                                                                                                                                                        
         [Node] Finished "select_boldmask_0", elapsed time 0.000703s.                                                                                                                                                            
241108-16:35:03,478 nipype.workflow ERROR:                                                                                                                                                                                       
         Node select_boldmask_1 failed to run on host ccs-ws01.                                                                                                                                                                  
241108-16:35:03,478 nipype.workflow ERROR:                                                                                                                                                                                       
         Saving crash info to /scratch/data/fmristroke/derivatives/sub-M2121/log/20241108-163432_19f04bdd-ca0c-48b9-a9dd-42c5b5b7560c/crash-20241108-163503-ehinger-select_boldmask_1-89490208-3e59-46d6-873f-0006f98a13a2.txt   
Traceback (most recent call last):                                                                                                                                                                                               
  File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node                                                                                                                                                                                                      
    result["result"] = node.run(updatehash=updatehash)                                                                                                                                                                                                        
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                        
  File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/pipeline/engine/nodes.py", line 527, in run                                                                                                                              
    result = self._run_interface(execute=True)                                                                                                                                                                                                                
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                
  File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/pipeline/engine/nodes.py", line 645, in _run_interface                                                                                                                   
    return self._run_command(execute)                                                                                                                                                                                                                         
           ^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                         
  File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/pipeline/engine/nodes.py", line 771, in _run_command                                                                                                                     
    raise NodeExecutionError(msg)                                                                                                                                                                                                                             nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node select_boldmask_1.                                                                                                                                                     
                                                                                                                                                                                                                                                              
Traceback:                                                                                                                                                                                                                                                    
        Traceback (most recent call last):                                                                                                                                                                                                                    
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/core.py", line 400, in run                                                                                                                       
            outputs = self.aggregate_outputs(runtime)                                                                                                                                                                                                         
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                         
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/base/core.py", line 429, in aggregate_outputs                                                                                                         
            predicted_outputs = self._list_outputs()  # Predictions from _list_outputs                                                                                                                                                                                                        ^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                          
          File "/home/ehinger/micromamba/envs/fmristroke/lib/python3.11/site-packages/nipype/interfaces/utility/base.py", line 412, in _list_outputs                                                                                                          
            out = np.array(self.inputs.inlist, dtype=object)[                                                                                                                                                                                                 
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                 
        IndexError: index 1 is out of bounds for axis 0 with size 1                                                                                                                                                                                           
                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                              241108-16:35:04,521 nipype.workflow INFO:                                                                                                                                                                                                                     
         [Node] Setting-up "_acc_msk_tfm0" in "/tmp/fmristroke/fmristroke_0_1_wf/single_subject_M2112_wf/lesion_preproc_ses_262_task_rest_run_10_space_T1w_desc_preproc_wf/lesion_confounds_wf/acc_msk_tfm/mapflow/_acc_msk_tfm0".                            
241108-16:35:04,522 nipype.workflow INFO:

**Comment by alixlam** [comment]
Ooh I see, I fixed this error in my last commit today, sorry about that. You actually should have all the derivatives in your output folder, I just forgot to stop some part of the processing for the bold runs that are too short so it raises some errors that have no impact on the desired outputs. These error messages are fixed in the  newest version of the code.

**Comment by alixlam** [comment]
Nice ! thank you ! I fixed the issue about the boilerplate, so no more error message. 
Thank you for trying on the latest fmriprep version ... It is hard to test all the version since new versions come out quite regularly. I added some info in the docs about fmriprep version, for now I recommend using versions prior to 24.0.0. 

Closing this issue, thank you again for your time and valuable feedbacks !
